src/renderer/entities/asset/lib/balanceService.ts

```diff
@@ -127,259 +149,248 @@ export const useBalance = (): IBalanceService => {
+        // }
+
+        return acc;
+      }, []);
+
+      insertNewBalances(accountIds, chain.chainId, newBalances);
     });
   };
 
   const subscribeStatemineAssetsChange = (
     accountIds: AccountId[],
     chain: ExtendedChain,
-    relaychain: ExtendedChain | undefined,
-    asset: Asset,
-  ) => {
-    const statemineAssetId = (asset?.typeExtras as StatemineExtras).assetId;
+    assets: Asset[],
+    relaychain?: ExtendedChain,
+  ): UnsubscribePromise => {
     const api = chain.api;
-    if (!api) return;
+    if (!api || !assets.length) return Promise.resolve(noop);
 
-    const addresses = accountIds.map((accountId) => toAddress(accountId, { prefix: chain.addressPrefix }));
+    const assetsMap = assets.reduce<[string, Address][]>((acc, asset) => {
+      accountIds.forEach((accountId) => {
+        acc.push([getAssetId(asset), toAddress(accountId, { prefix: chain.addressPrefix })]);
+      });
+
+      return acc;
+    }, []);
 
-    return api.query.assets.account.multi(
-      addresses.map((a) => [statemineAssetId, a]),
-      (data: any[]) => {
-        data.forEach(async (accountInfo: Option<any>, i) => {
-          try {
-            const free = accountInfo.isNone ? '0' : accountInfo.unwrap().balance.toString();
-            const balance = {
-              accountId: accountIds[i],
-              chainId: chain.chainId,
-              assetId: asset.assetId.toString(),
-              verified: true,
-              free: free.toString(),
-              frozen: (0).toString(),
-              reserved: (0).toString(),
-            };
-
-            const existingBalance = await balanceStorage.getBalance(
-              balance.accountId,
-              balance.chainId,
-              balance.assetId,
-            );
-            if (!existingBalance) {
-              await addBalance(balance);
-            } else if (balance.free !== existingBalance.free) {
-              await updateBalance(balance);
-            }
-
-            if (relaychain?.api && isLightClient(relaychain)) {
-              const storageKey = api.query.assets.account.key(statemineAssetId, addresses[i]);
-              runValidation(
-                relaychain.api,
-                api,
-                storageKey,
-                accountInfo,
-                () => setBalanceIsValid(balance, true),
-                () => setBalanceIsValid(balance, false),
-              );
-            }
-          } catch (e) {
-            console.warn(e);
-          }
+    return api.query.assets.account.multi(assetsMap, (data: any[]) => {
+      const newBalances = data.reduce((acc, accountInfo, index) => {
+        const free = accountInfo.isNone ? '0' : accountInfo.unwrap().balance.toString();
+        acc.push({
+          accountId: toAccountId(assetsMap[index][1]),
+          chainId: chain.chainId,
+          assetId: assets[index].assetId.toString(),
+          verified: true,
+          frozen: (0).toString(),
+          reserved: (0).toString(),
```

Comment:
Just noticed we're not taking account `accountInfo.status` which might be the following
``` kotlin
enum class AccountStatus {
        Liquid, Frozen, Blocked
    }
```
In case status is Frozen or Blocked we should put balance not into `acc.free` but to `acc.frozen`
Not sure if we need to fix it now, but at least we can create a task for it

Reply:
[Task](https://app.clickup.com/t/869339q2r)